### PR TITLE
[KYUUBI #6668] Fix kyuubi batch state abnormal

### LINKED ISSUE #6668: [Bug] Kyuubi batch state abnormal
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

1. received the delete batch rest api
```
2024-09-04 16:05:01.441 user=b_mac(auth:KYUUBI_INTERNAL)	ip=10.174.193.202	proxyIp=null	method=DELETE	uri=/api/v1/batches/eec76349-895f-479d-b9c2-58d50aefd866	params=null	protocol=HTTP/1.1	status=200
```
2. close session:
```
2024-09-04 16:05:01.043 INFO org.apache.kyuubi.session.KyuubiSessionManager: b_mac's session with SessionHandle [eec76349-895f-479d-b9c2-58d50aefd866]/apollorno/6581cbd5-8ee6-4cdd-9354-bcfcea2eaec3-6464_6581cbd5-8ee6-4cdd-9354-bcfcea2eaec3-6464.90c0b328-930f-11ed-a1eb-0242ac120002.0_20240904230444.stm is closed, current opening sessions 181
```
3.  batch operation state change
```
/var/log/hadoop/kyuubi/kyuubi-stdout.log.1:2024-09-04 16:05:01.397 INFO org.apache.kyuubi.operation.BatchJobSubmission: Processing b_mac's query[23f8338f-4935-465f-8371-f11459fccbd4]: PENDING_STATE -> ERROR_STATE, time taken: 1.725491101397E9 seconds
```

Abnormal:
1. why close session before operation?
2. why final operation state is `ERROR` but not `CANCELED`?
3. the pod was not deleted because it was not created yet, and then the pod created and running successfully. 

### Affects Version(s)

master

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by turboFei**
# :mag: Description
## Issue References 🔗

This pull request fixes #6668

## Describe Your Solution 🔧

1. when failed to kill the batch, check the current application info
2. if the application state is UNKNOWN(less than submit timeout) or NOT_FOUND, mark the batch state to CANCELED 
3. If the k8s pod added after the batch marked as CANCELED, delete the pod

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by turboFei** [comment]
cc @pan3793 @ulysses-you @zwangsheng

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala [hunk 1]
```diff
@@ -69,11 +72,16 @@ class KubernetesApplicationOperation extends ApplicationOperation with Logging {
...
   private var expireCleanUpTriggerCacheExecutor: ScheduledExecutorService = _
 
+  private var cleanupCanceledAppPodExecutor: ThreadPoolExecutor = _
+
   private def getOrCreateKubernetesClient(kubernetesInfo: KubernetesInfo): KubernetesClient = {
     checkKubernetesInfo(kubernetesInfo)
     kubernetesClients.computeIfAbsent(kubernetesInfo, kInfo => buildKubernetesClient(kInfo))
   }
 
+  private lazy val metadataManager = KyuubiServer.kyuubiServer.backendService
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala:82]: maybe `lazy val` => `def`, to tolerate the unexpected starting order

**Review by pan3793 — APPROVED** [review]
LGTM

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala (hunk 1 above)
Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala]: done

**Comment by turboFei** [comment]
thanks, merged to 1.10.0/1.9.3

**Pull request closed without merging**